*** ppl-0.9/interfaces/Prolog/GNU/README.gprolog	1970-01-01 01:00:00.000000000 +0100
--- ppl-0.9-patched/interfaces/Prolog/GNU/README.gprolog	2007-02-18 11:37:53.000000000 +0100
***************
*** 0 ****
--- 1,36 ----
+ The GNU Prolog interface to the PPL is available both as a ``PPL
+ enhanced'' GNU Prolog interpreter and as a library that can be linked
+ to GNU Prolog programs.  Only GNU Prolog version 1.2.19 and later
+ versions are supported.
+ 
+ So that it can be used with the GNU Prolog PPL interface (and, for
+ that matter, with any foreign code), the GNU Prolog installation must
+ be configured with the `--disable-regs' option.
+ 
+ 
+ The `ppl_gprolog' Executable
+ ----------------------------
+ 
+ The `ppl_gprolog' executable is simply the GNU Prolog interpreter with
+ the Parma Polyhedra Library linked in.  The only thing you should do
+ to use the library is to call `ppl_initialize/0' before any other PPL
+ predicate and to call `ppl_finalize/0' when you are done with the
+ library.
+ 
+ 
+ Linking the Library To GNU Prolog Programs
+ ------------------------------------------
+ 
+ In order to allow linking GNU Prolog programs to the PPL, the
+ following files are installed in the directory `<prefix>/lib/ppl',
+ where <prefix> is the root of your installation (`/usr' or
+ `/usr/local' in most cases): `ppl_gprolog.pl' contains the required
+ foreign declarations; `libppl_gprolog.*' contain the executable code
+ for the GNU Prolog interface in various formats (static library,
+ shared library, libtool library).  If your GNU Prolog program is
+ constituted by, say, `source1.pl' and `source2.pl' and you want to
+ create the executable `myprog', your compilation command may look like
+ 
+     gplc -o myprog <prefix>/lib/ppl/ppl_gprolog.pl source1.pl source2.pl \
+       -L '-L<prefix>/lib/ppl -lppl_gprolog -L<prefix>/lib \
+       -lppl -lgmpxx -lgmp -lstdc++'
*** ppl-0.9/interfaces/Prolog/SWI/README.swiprolog	1970-01-01 01:00:00.000000000 +0100
--- ppl-0.9-patched/interfaces/Prolog/SWI/README.swiprolog	2007-02-18 11:38:09.000000000 +0100
***************
*** 0 ****
--- 1,33 ----
+ The SWI-Prolog interface to the PPL is available both as a statically
+ linked module or as a dynamically linked one.  Only SWI-Prolog version
+ 5.6.0 and later versions are supported.
+ 
+ 
+ The `ppl_pl' Executable
+ -----------------------
+ 
+ The `ppl_pl' executable is simply the SWI-Prolog shell with the Parma
+ Polyhedra Library statically linked: from within `ppl_pl' all the
+ services of the library are available without further action.
+ 
+ 
+ Loading the SWI-Prolog Interface Dynamically
+ --------------------------------------------
+ 
+ In order to dynamically load the library from SWI-Prolog you should
+ simply load `<prefix>/lib/ppl/ppl_swiprolog.pl', where <prefix> is the
+ root of your installation (`/usr' or `/usr/local' in most cases).
+ This will invoke `ppl_initialize/0' and `ppl_finalize/0'
+ automatically.  Alternatively, you can load the library directly with
+ 
+     :- load_foreign_library('<prefix>/lib/ppl/libppl_swiprolog').
+ 
+ This will call `ppl_initialize/0' automatically.  Analogously,
+ 
+     :- unload_foreign_library('<prefix>/lib/ppl/libppl_swiprolog').
+ 
+ will, as part of the unload process, invoke `ppl_finalize/0'.
+ 
+ Notice that, for dynamic linking to work, you should have configured
+ the library with the `--enable-shared' option (on most platforms this
+ is the default).
*** ppl-0.9/interfaces/Prolog/YAP/README.yap	1970-01-01 01:00:00.000000000 +0100
--- ppl-0.9-patched/interfaces/Prolog/YAP/README.yap	2007-02-18 11:38:21.000000000 +0100
***************
*** 0 ****
--- 1,11 ----
+ The YAP Prolog interface to the PPL is available as a dynamically
+ linked module.  Only YAP version 5.1.1 and following are supported.
+ 
+ In order to dynamically load the library from YAP you should simply
+ load `<prefix>/lib/ppl/ppl_yap.pl', where <prefix> is the root of your
+ installation (`/usr' or `/usr/local' in most cases).  This will invoke
+ `ppl_initialize/0' automatically; it is the programmer's
+ responsibility to call `ppl_finalize/0' when the PPL library is no
+ longer needed.  Notice that, for dynamic linking to work, you should
+ have configured the library with the `--enable-shared' option (on most
+ platforms this is the default).
*** ppl-0.9/Watchdog/README.doc	1970-01-01 01:00:00.000000000 +0100
--- ppl-0.9-patched/Watchdog/README.doc	2007-02-18 11:37:34.000000000 +0100
***************
*** 0 ****
--- 1,5 ----
+ Use the Source, Luke!
+ 
+ Seriously, for this release of the Parma Watchdog Library,
+ documentation is only available in the sources, in Doxygen format.
+ Proper manuals will be included in future releases.
diff -ur ppl-0.9/doc/Makefile.in ppl-0.9-patched/doc/Makefile.in
--- ppl-0.9/doc/Makefile.in	2006-03-12 09:11:41.000000000 +0100
+++ ppl-0.9-patched/doc/Makefile.in	2007-06-12 20:41:45.000000000 +0200
@@ -710,7 +710,7 @@
 	for file in $(srcdir)/*.txt $(srcdir)/README.doc; do \
 	  $(INSTALL_DATA) $$file $(DESTDIR)$(txtdir)/`basename $$file`; \
 	done
-	for file in BUGS COPYING CREDITS ChangeLog NEWS README README.configure ; do \
+	for file in BUGS COPYING CREDITS ChangeLog NEWS README README.configure TODO ; do \
 	  $(INSTALL_DATA) $(srcdir)/../$$file $(DESTDIR)$(txtdir)/$$file; \
 	done
 
@@ -719,7 +719,7 @@
 
 uninstall-txt:
 	rm -f $(DESTDIR)$(txtdir)/*.txt $(DESTDIR)$(txtdir)/README.doc
-	for file in BUGS COPYING CREDITS ChangeLog NEWS README README.configure ; do \
+	for file in BUGS COPYING CREDITS ChangeLog NEWS README README.configure TODO ; do \
 	  rm -f $(DESTDIR)$(txtdir)/$$file; \
 	done
 
diff -ur ppl-0.9/doc/definitions.dox ppl-0.9-patched/doc/definitions.dox
--- ppl-0.9/doc/definitions.dox	2006-02-12 10:13:39.000000000 +0100
+++ ppl-0.9-patched/doc/definitions.dox	2007-07-02 14:22:03.000000000 +0200
@@ -2286,7 +2286,7 @@
 \anchor BHRZ03b
 R.&nbsp;Bagnara, P.&nbsp;M. Hill, E.&nbsp;Ricci, and E.&nbsp;Zaffanella.
   Precise widening operators for convex polyhedra.
-  Quaderno 312, Dipartimento di Matematica, Universit‡ di Parma,
+  Quaderno 312, Dipartimento di Matematica, Universit√† di Parma,
   Italy, 2003.
   Available at <A HREF="http://www.cs.unipr.it/Publications/">http://www.cs.unipr.it/Publications/</A>.
 
@@ -2299,7 +2299,7 @@
 R.&nbsp;Bagnara, P.&nbsp;M. Hill, and E.&nbsp;Zaffanella.
   A new encoding and implementation of not necessarily closed convex
   polyhedra.
-  Quaderno 305, Dipartimento di Matematica, Universit‡ di Parma,
+  Quaderno 305, Dipartimento di Matematica, Universit√† di Parma,
   Italy, 2002.
   Available at <A HREF="http://www.cs.unipr.it/Publications/">http://www.cs.unipr.it/Publications/</A>.
 
@@ -2315,8 +2315,8 @@
   the 1st CoLogNet Workshop on Component-based Software Development and
   Implementation Technology for Computational Logic Systems</i>, pages 147-153,
   Madrid, Spain, 2002.
-  Published as TR Number CLIP4/02.0, Universidad PolitÈcnica de
-  Madrid, Facultad de Inform·tica.
+  Published as TR Number CLIP4/02.0, Universidad Polit√©cnica de
+  Madrid, Facultad de Inform√°tica.
 
 </DD>
 
@@ -2353,7 +2353,7 @@
 \anchor BHZ04
 R.&nbsp;Bagnara, P.&nbsp;M. Hill, and E.&nbsp;Zaffanella.
   Widening operators for powerset domains.
-  Quaderno 349, Dipartimento di Matematica, Universit‡ di Parma,
+  Quaderno 349, Dipartimento di Matematica, Universit√† di Parma,
   Italy, 2004.
   Available at <A HREF="http://www.cs.unipr.it/Publications/">http://www.cs.unipr.it/Publications/</A>.
 
@@ -2365,7 +2365,7 @@
 \anchor BJT99
 F.&nbsp;Besson, T.&nbsp;P. Jensen, and J.-P. Talpin.
   Polyhedral analysis for synchronous languages.
-  In A.&nbsp;Cortesi and G.&nbsp;FilÈ, editors, <i>Static Analysis:
+  In A.&nbsp;Cortesi and G.&nbsp;Fil√©, editors, <i>Static Analysis:
   Proceedings of the 6th International Symposium</i>, volume 1694 of <i>Lecture
   Notes in Computer Science</i>, pages 51-68, Venice, Italy, 1999.
   Springer-Verlag, Berlin.
@@ -2393,7 +2393,7 @@
 R.&nbsp;Bagnara, E.&nbsp;Ricci, E.&nbsp;Zaffanella, and P.&nbsp;M. Hill.
   Possibly not closed convex polyhedra and the Parma Polyhedra
   Library.
-  Quaderno 286, Dipartimento di Matematica, Universit‡ di Parma,
+  Quaderno 286, Dipartimento di Matematica, Universit√† di Parma,
   Italy, 2002.
   See also \ref BRZH02c "[BRZH02c]". Available at
   <A HREF="http://www.cs.unipr.it/Publications/">http://www.cs.unipr.it/Publications/</A>.
@@ -2541,7 +2541,7 @@
 E.&nbsp;Gawrilow and M.&nbsp;Joswig.
   <tt>polymake</tt>: a framework for analyzing convex polytopes.
   In G.&nbsp;Kalai and G.&nbsp;M. Ziegler, editors, <i>Polytopes -
-  Combinatorics and Computation</i>, pages 43-74. Birkh‰user, 2000.
+  Combinatorics and Computation</i>, pages 43-74. Birkh√§user, 2000.
 
 </DD>
 
@@ -2562,10 +2562,10 @@
 <DD>
 \anchor Hal79
 N.&nbsp;Halbwachs.
-  <i>DÈtermination Automatique de Relations LinÈaires
-  VÈrifiÈes par les Variables d'un Programme</i>.
-  ThËse de 3Ëme cycle d'informatique,
-  UniversitÈ scientifique et mÈdicale de Grenoble, Grenoble, France, March
+  <i>D√©termination Automatique de Relations Lin√©aires
+  V√©rifi√©es par les Variables d'un Programme</i>.
+  Th√®se de 3√®me cycle d'informatique,
+  Universit√© scientifique et m√©dicale de Grenoble, Grenoble, France, March
   1979.
 
 </DD>
@@ -2725,7 +2725,7 @@
 F.&nbsp;Masdupuy.
   <i>Array Indices Relational Semantic Analysis Using Rational Cosets
   and Trapezoids</i>.
-  ThËse d'informatique, …cole Polytechnique, Palaiseau, France,
+  Th√®se d'informatique, √âcole Polytechnique, Palaiseau, France,
   December 1993.
 
 </DD>
diff -ur ppl-0.9/NEWS ppl-0.9-patched/NEWS
--- ppl-0.9/NEWS	2006-03-05 18:31:24.000000000 +0100
+++ ppl-0.9-patched/NEWS	2007-07-02 14:22:29.000000000 +0200
@@ -1006,10 +1006,10 @@
    of the widenings introduced in
 
      N. Halbwachs.
-     DÈtermination Automatique de Relations LinÈaires
-     VÈrifiÈes par les Variables d'un Programme.
-     ThËse de 3Ëme cicle d'informatique,
-     UniversitÈ scientifique et mÈdicale de Grenoble,
+     D√©termination Automatique de Relations Lin√©aires
+     V√©rifi√©es par les Variables d'un Programme.
+     Th√®se de 3√®me cicle d'informatique,
+     Universit√© scientifique et m√©dicale de Grenoble,
      Grenoble, France, March 1979.
 
    and described in
diff -ur ppl-0.9/TODO ppl-0.9-patched/TODO
--- ppl-0.9/TODO	2006-03-05 22:50:55.000000000 +0100
+++ ppl-0.9-patched/TODO	2007-07-02 14:22:40.000000000 +0200
@@ -37,7 +37,7 @@
 - Reimplement the ask-and-tell generic construction.
 - Suppose we are only interested in integral solutions (as is the
   case, e.g., in most termination analyses).  Then cutting-plane methods
-  (Gomory, Chv·tal, ...) allow to shrink polyhedra still not losing any
+  (Gomory, Chv√°tal, ...) allow to shrink polyhedra still not losing any
   integral solution.
   See http://www.cs.unipr.it/ppl/Documentation/bibliography#NemhauserW88
 
